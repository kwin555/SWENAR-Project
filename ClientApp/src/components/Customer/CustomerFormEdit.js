import React, { Component } from "react";

class CustomerFormEdit extends Component {
  //default state of app is empty name and number
  constructor(props) {
    super(props);

    this.state = {
      Name: "",
      number: "",
      error: undefined,
    };
  }

  //onchange handler for text input
  onChange = (e) => {
    this.setState({
      [e.target.name]: e.target.value,
    });
  };

  //call to back end to edit an existing name
  submitToPUT = () => {
    let { id } = this.props;
    id = parseInt(id);
    const axios = require("axios");
    const { Name, number } = this.state;
    this.props.handleNameNumberchange(Name, number);

    axios
      .put(`/api/customer/${id}`, {
        id,
        Name,
        Number: number,
      })
      .then(function (response) {
        console.log(response);
      })
      .catch(function (error) {
        this.setState({ error: error });
      });
  };

  render = () => {
    return (
      <div>
        <form>
          <label htmlFor='Name'>Enter new name: </label>
          <input id='Name' name='Name' type='text' onChange={this.onChange} />

          <label htmlFor='Number'>Enter new number:</label>
          <input
            id='number'
            name='number'
            type='text'
            onChange={this.onChange}
          />
        </form>
        <div>
          <button onClick={this.submitToPUT}>Save</button>
        </div>
      </div>
    );
  };
}

export default CustomerFormEdit;
