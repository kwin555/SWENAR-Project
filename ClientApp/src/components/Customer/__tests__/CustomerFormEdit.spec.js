import React from "react";
import CustomerFormEdit from "../CustomerFormEdit";
import { shallow } from "enzyme";
import axios from "axios";

describe("CustomerForm tests", () => {
  it("shows customer forms with things rendered", () => {
    const props = {
      Name: "",
      number: "",
    };
    const wrapper = shallow(<CustomerFormEdit {...props}></CustomerFormEdit>);
    expect(wrapper).toMatchSnapshot();
  });

  it("Should call setState on input", () => {
    const mockChange = {
      target: {
        name: "input",
        value: "test 123",
      },
    };

    const expected = {
      Name: "",
      number: "",
      input: "test 123",
      error: undefined,
    };

    const props = {
      Name: "",
      number: "",
    };

    const wrapper = shallow(<CustomerFormEdit {...props} />);

    wrapper.instance().onChange(mockChange);

    expect(wrapper.state()).toEqual(expected);
  });

  it("Should call submitToPUT", () => {
    const props = {
      Name: "",
      number: "",
      handleNameNumberchange: jest.spyOn(axios, "put"),
    };

    const mockChange = {
      target: {
        name: "input",
        value: "test 123",
      },
    };

    const wrapper = shallow(<CustomerFormEdit {...props} />);

    wrapper.instance().onChange(mockChange);
    wrapper.instance().submitToPUT();

    wrapper.find("button").simulate("click");
    expect(props.handleNameNumberchange).toHaveBeenCalled();
    wrapper.unmount();
  });
});
