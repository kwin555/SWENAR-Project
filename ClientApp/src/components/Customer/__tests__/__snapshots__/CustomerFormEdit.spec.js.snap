// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CustomerForm tests shows customer forms with things rendered 1`] = `
<div>
  <form>
    <label
      htmlFor="Name"
    >
      Enter new name: 
    </label>
    <input
      id="Name"
      name="Name"
      onChange={[Function]}
      type="text"
    />
    <label
      htmlFor="Number"
    >
      Enter new number:
    </label>
    <input
      id="number"
      name="number"
      onChange={[Function]}
      type="text"
    />
  </form>
  <div>
    <button
      onClick={[Function]}
    >
      Save
    </button>
  </div>
</div>
`;
